static/js/lib/api.js

```diff
@@ -6,6 +6,92 @@ import R from "ramda"
 
 import { S, parseJSON, filterE } from "./sanctuary"
 
+export const booksRequest = (force = false) => ({
+  url: "/api/library/books/",
+  update: {
+    books: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    // The response is a list of serialized books. This turns it into an object mapping
+    // the book id to the full serialized object: {[book.id]: [book]}
+    return {
+      books: R.fromPairs(
+        responseJson.map((book) => (
+          [book.id, book]
+        ))
+      )
+    }
+  },
+  force
+})
```

Comment:
`update` is an object that maps an entity ID (`books` in this case) to a function. That function is given the previous value of `books` in the redux state and the value of the `books` key returned from `transform` (or if `transform` isn't defined, the `books` key in the raw response data). 

`transform` takes the response data and modifies it so that it has keys that match the `update` object.

In this case, we only want the state to reflect the most recent results from the books endpoint, so we just return `next`.

Reply:
this is where we would define logic for merging new data in with old data, if we needed to?

Reply:
Yup, we would do that in `update`. If, for example, we were defining the endpoint for an infinite-scrollable post list in Open, we would append the `next` value to `prev` instead of just returning next

Reply:
cool, makes sense!